swimlane/core/adapters/task.py

```diff
@@ -0,0 +1,55 @@
+from swimlane.core.resolver import SwimlaneResolver
+from swimlane.core.cache import check_cache
+from swimlane.core.resources.task import Task
+from swimlane.utils import one_of_keyword_only
+
+
+class TaskAdapter(SwimlaneResolver):
+    """Handles retreival of Swimlane Task Resources and execution of tasks against records."""
+    
+    @check_cache(Task)
+    @one_of_keyword_only('id', 'name')
+    def get(self, key, value):
+        """Get a single task by id or name"""
+        task_list = self.list()
+        if key == 'id':
+            for task in task_list:
+                if value == task.id:
+                    return task
+            raise ValueError('No task with id "{value}"'.format(value=value))
+        if key == 'name':
+            for task in task_list:
+                if value == task.name:
+                    return task
+            raise ValueError('No task with name "{value}"'.format(value=value))
+
+    def list(self):
+        """Retrieve list of all tasks.
+
+        Returns:
+            :class:`list` of :class:`~swimlane.core.resources.task.Task`: List of all tasks.
+        """
+        response = self._swimlane.request('get', 'task/light')
+        return [self._get_full(item.get('id')) for item in response.json()]
```

Comment:
Enriching all seems like an expensive action. Maybe identify by name and then enrich? By id is already enriched.

Reply:
Yeah, I can see your point. I'll rework these two functions.

I used the list function of the App adapter as an example of how to look up a task by name. But I can come up with a better way.

https://github.com/swimlane/swimlane-python/blob/868c9d8da86d031243300b4b9263aa6c72728341/swimlane/core/adapters/app.py#L38-L55

Reply:
The point I am making is that it is better to enrich just one - after you identify that one by name, instead of enriching all even if you don't need them at all.

Reply:
I've made the change to enrich just one. I profiled the method before and after using line_profiler, and the new way was ~28.5 % faster. This is only on a server with 3 tasks, so I can imagine it would be an even bigger improvement with one that had many more.

Reply:
To clarify, I made get a task by name only enrich one. I left the list function there so it can be used to pull a full list and iterate on the objects if someone needs it. It would work well with dumping out properties of Task, like script if someone needed to do it. A generator function would be more memory efficient.